Makefile

```diff
@@ -73,9 +73,15 @@ clean:
 	go clean -cache -testcache ./...
 
 # Only build image with lint-free code. CI will fail on lint errors.
-image: lint
+image: lint 
 	@if [ $${SKIP_BUILD:-false} = false ] ; then \
-		podman build -t $(IMAGE_TAG) . ; \
+		cp Dockerfile Dockerfile.local ; \
+		if [ -f local_dockerfile_changes ] ; then \
+			while read -r line; do  \
+				sed -i "s~$$line~g" Dockerfile.local ; \
+			done < local_dockerfile_changes ; \
```

Comment:
This feels very unconventional and kind of confusing to me. Is there a way to use a standard `.patch` file or even commit the Dockerfile.local to be used for local builds only? Meaning, can we use some kind of CI environment variable to override the default FROM image instead?

Reply:
> This feels very unconventional and kind of confusing to me. Is there a way to use a standard `.patch` file or even commit the Dockerfile.local to be used for local builds only? Meaning, can we use some kind of CI environment variable to override the default FROM image instead?

This is fundamentally what ART does now.  I would be onboard with a "patching" system but against checking in Dockerfile.local since needing to maintain 2 files is fraught with error IMO.

Reply:
You could use `diff` and `patch` to create a diff and apply it to Dockerfile. 
Also suggest you make Dockerfile.local a proper make target:
```
Dockerfile.local: Dockerfile Dockerfile.local.diff
	patch -i Dockerfile.local.diff -o Dockerfile.Local Dockerfile

image: Dockerfile.local
```

If the `patch` command isn't available on the CI boxes, you can use `git apply` to do the same thing - you'll have to check the man pages for flags and whatnot.